cli/verifier/versions_match.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package verifier
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/edgelesssys/contrast/internal/kuberesource"
+
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	applymetav1 "k8s.io/client-go/applyconfigurations/meta/v1"
+)
+
+type identifier struct {
+	podRole string
+	image   string
+}
+
+var versionedResources = []identifier{
+	{image: "coordinator", podRole: "coordinator"},
+	{image: "contrast-initializer"},
+	{image: "contrast-service-mesh"},
+}
+
+func matchesResource(image, podRole string) bool {
+	for _, r := range versionedResources {
+		if r.image == image {
+			if podRole == "" || r.podRole == podRole {
```

Comment:
If I add a pod with a role other than coordinator (we don't have that right now, but might), it will be skipped, right?

Reply:
You mean: if I add a container with name "coordinator"  in a pod with non-coordinator role, it will be skipped? If so, then: Yes, I thought that was the intention behind your previous comment about the name "coordinator" not being a sufficient indicator

Reply:
I find the logic as implemented a bit hard to follow. What we want is this, right?

```go
func needsImageVersionCheck(containerName string, podRole string) {
  if slices.Contains([]string{"contrast-initializer", "contrast-service-mesh"}, containerName) {
    return true
  }
  if containerName == "coordinator" && podRole == "coordinator" {
    return true
  }
  return false
}
```